packit_service/worker/psbugzilla.py

```diff
@@ -0,0 +1,173 @@
+# MIT License
+#
+# Copyright (c) 2020 Red Hat, Inc.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+""" The file is called psbugzilla to avoid conflicts with python-bugzilla module. """
+
+import logging
+from os import getenv
+from tempfile import TemporaryFile
+from typing import List, Tuple
+from xmlrpc.client import Fault
+
+import backoff
+from bugzilla import Bugzilla as XMLRPCBugzilla
+from requests import get
+
+from ogr.abstract import PullRequest
+from ogr.services.pagure import PagureService
+
+
+class Bugzilla:
+    """ TODO """
+
+    def __init__(self, api_key: str = None):
+        self.logger = logging.getLogger(__name__)
+        self._api_key = api_key or getenv("BUGZILLA_API_KEY")
+        self._url = None
+        self._bzapi = None
+
+    @property
+    def url(self):
+        if self._url is None:
+            if getenv("DEPLOYMENT") == "prod":
+                self._url = "https://bugzilla.redhat.com"
+            else:
+                self._url = "https://partner-bugzilla.redhat.com"
```

Comment:
Instead of linking these URLs to the deployment name and hard-coding them, make them configurable.

Reply:
Why? I don't know of any other bugzilla instances that could be used in production/staging, do you?

Where? in packit-service.yaml?

Reply:
> Why? I don't know of any other bugzilla instances that could be used in production/staging, do you?

For flexibility: so that we can point any Packit Service instance to any Bugzilla instance, independent of what we call the current environment.

> 
> Where? in packit-service.yaml?

Yes.